lib/swm050/gpio.c

```diff
@@ -187,11 +183,129 @@ is enabled, GPIO and AF of the SWD pins will be both unavailable.
 */
 void gpio_sel_swd(bool en)
 {
-	if (en) {
-		SWD_SEL = 1;
-	} else {
-		SWD_SEL = 0;
+	SWD_SEL = en;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the pins as external interrupts, rather than normal GPIO
+
+Enable interrupts on the selected pins.  If you want to quickly
+switch on and off interrupts, use gpio_int_mask() after calling this.
+
+@param[in] gpios Pin identifiers @ref gpio_pin_id
+	     If multiple pins are to be changed, use bitwise OR '|' to separate
+	     them.
+*/
+void gpio_int_en(uint16_t gpios)
+{
+	for (uint8_t i = 0; 0 <= 9; ++i) {
+		if (gpios & (1 << i)) {
+			nvic_enable_irq(NVIC_GPIOA0_IRQ + i);
```

Comment:
traditionally on other families, we've only enabled the irqs in the peripheral blocks, and user code hsa been responsible for nvic layer stuff.  I'd like to see that either preserved, or a discussion on it across the whole tree.

Reply:
Fixed.